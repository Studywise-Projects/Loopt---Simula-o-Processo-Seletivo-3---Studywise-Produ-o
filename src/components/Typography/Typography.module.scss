@import '@/styles/_global.scss';

.typography {
  color: $black;
  line-height: 1.5;
}

.header {
  color: $white;
  font-size: 24px;
  font-weight: 700;

  @media only screen and (min-width: 1024px) {
    font-size: 32px;
  }
}

.titleBlack {
  font-size: 24px;
  font-weight: 700;

  @media only screen and (min-width: 1024px) {
    font-size: 32px;
  }
}

.titleBlue {
  color: $primary-color;
  font-size: 24px;
  font-weight: 700;

  @media only screen and (min-width: 1024px) {
    font-size: 32px;
  }
}

.subtitle {
  font-size: 20px;
  font-weight: 700;
}

.subtitle2 {
  font-size: 14px;
  font-weight: 700;
}

.body {
  font-size: 14px;
  font-weight: 400;
}

.caption {
  color: $gray;
  font-size: 14px;
  font-weight: 400;
}

.error {
  color: red;
  font-size: 14px;
  font-weight: 400;
}
