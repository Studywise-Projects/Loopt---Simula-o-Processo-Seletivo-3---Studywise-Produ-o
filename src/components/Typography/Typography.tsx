import { Typography as TypographyMui } from '@mui/material';
import classNames from 'classnames';
import styles from './Typography.module.scss';

interface ITypography {
  variant: 'header' | 'title' | 'subtitle' | 'subtitle2' | 'body' | 'caption';
  text: string;
}

function Typography({ variant, text }: ITypography) {
  const typographyStyles = classNames(styles.typography, {
    [styles[variant]]: variant,
  });

  return <TypographyMui className={typographyStyles}>{text}</TypographyMui>;
}

export default Typography;
