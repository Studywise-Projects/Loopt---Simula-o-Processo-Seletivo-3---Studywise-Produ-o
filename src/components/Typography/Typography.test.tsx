import { render } from '@testing-library/react';
import Typography from './Typography';

describe('Typography', () => {
  it('Should match the snapshot', () => {
    const { getByText } = render(<Typography variant='body' text='test' />);

    const element = getByText('test');
    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByText } = render(<Typography variant='body' text='test' />);

    const element = getByText('test');
    expect(element).toBeInTheDocument();
  });
});
