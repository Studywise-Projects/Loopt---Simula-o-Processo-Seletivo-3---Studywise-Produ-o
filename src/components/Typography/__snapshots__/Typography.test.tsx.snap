// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Should match the snapshot 1`] = `ShallowWrapper {}`;
